nifi-nar-bundles/nifi-cassandra-bundle/nifi-cassandra-processors/src/main/java/org/apache/nifi/processors/cassandra/QueryCassandra.java

```diff
@@ -130,6 +132,23 @@
             .defaultValue(AVRO_FORMAT)
             .build();
 
+    public static final PropertyDescriptor DATE_FORMAT_PATTERN = new PropertyDescriptor.Builder()
```

Comment:
TL;DR I think this should be called "Timestamp Format Pattern for JSON output":

I think (for now) we have to be pretty specific about the fields this property works upon. For example there is a separate bug (that should be written up as a Jira) where the CQL `DATE` type is not fully supported. For example, their `DATE` type returns a Cassandra-specific `LocalDate` class, which cannot currently be translated to an Avro schema, and in the current code (PR included) it drops through the JSON `instanceof Date` clause and is issued as `value.toString()`. So being a `java.util.Date` type lends itself to a Cassandra `TIMESTAMP` type, which is reflected in the unit tests. It's odd to me that they return a `java.util.Date` when (from [other sources](http://itdoc.hitachi.co.jp/manuals/3020/30203V0300e/BV030040.HTM), a vendor not the community) it appears they have a full-fledged `java.sql.Timestamp` object under the hood, yet we can only access a Date object so we can't get things like nanoseconds.

We should revisit the `DATE` and `TIME` datatypes under a separate Jira but since this only seems to apply to the `TIMESTAMP` type, I'm thinking we should name it as such. I believe the JsonRecordSetWriter does something similar (i.e. has properties for date, time, timestamp formats)

Reply:
Thanks @mattyb149 for the review.
I have updated the name/displayname according to your suggestion.
I agree that the time/date handling should be improved, currently the CQL `timestamp` works correctly, but I'm not sure about the `date` and `time` types, to be honest. The aim of this PR was to add the ability to change the timestamp's formatting in JSON output as with the current formatting it discards the milliseconds, but I did not want to change anything else.
I'll file a jira for this.